ml-agents/mlagents/trainers/model_saver/torch_model_saver.py

```diff
@@ -89,7 +89,21 @@ def _load_model(
         policy = cast(TorchPolicy, policy)
 
         for name, mod in modules.items():
-            mod.load_state_dict(saved_state_dict[name])
+            try:
+                missing_keys, unexpected_keys = mod.load_state_dict(
+                    saved_state_dict[name], strict=False
+                )
+                if missing_keys:
+                    logger.warning(
+                        f"Did not find these keys {missing_keys} in checkpoint. Initializing"
+                    )
+                if unexpected_keys:
+                    logger.warning(
+                        f"Did not expect these keys {unexpected_keys} in checkpoint. Ignoring"
+                    )
+
+            except (KeyError, TypeError):
```

Comment:
Can you logger.debug the original error ?

Reply:
Instead of or in addition to this warning?